### ISSUE TITLE
Csc compiler error after upgrading to VS2015 Update 3 (order of import statements)

### ISSUE THREAD

**Issue by ianbattersby** [fsprojects/Paket#1789]
### Description

Please provide a succinct description of your issue.

### Repro steps

Please provide the steps required to reproduce the problem

1. Step A

2. Step B

### Expected behavior

Please provide a description of the behavior you expect.

### Actual behavior

Please provide a description of the actual behavior you observe. 

### Known workarounds

Please provide a description of any known workarounds.

### Related information 

* Operating system
* Branch
* .NET Runtime, CoreCLR or Mono Version
* Performance information, links to performance testing scripts

**Comment by smoothdeveloper** [comment]
@ianbattersby thanks for detailed error report.

I have no clue about `Microsoft.CodeDom.Providers.DotNetCompilerPlatform` and `Microsoft.Net.Compilers` packages nor what makes Web API projects.

Is there a logical way paket should ideally figure out the order?

Does one nuget package depends on the other?

Would you mind making a PR with a failing test case added in integration tests project? This will help someone looking further at the bug and will make sure it is nailed under a test.

https://github.com/fsprojects/Paket/tree/master/integrationtests

Thanks!

**Comment by matthid** [comment]
Well the props files should always be first. Idk if the order of the props files actually matters, but the problem here is that the targets file is imported before the props files...

The testcase would in fact help here if possible.

**Comment by forki** [comment]
@ianbattersby you can just mail me the project - I will take a look

**Comment by ianbattersby** [comment]
@forki Should be able to do a repro, bear with - mad day ;)

**Comment by liboz** [comment]
@forki do you still need a repro?

**Comment by forki** [comment]
yeah I think it's still open

**Comment by ianbattersby** [comment]
Let me check that with the newer Net.Compilers package this is still an issue.

**Comment by liboz** [comment]
It still applies to me. I'll try to make a repro later today.

**Comment by liboz** [comment]
@forki @ianbattersby I made a fork of Paket with an example of this issue: https://github.com/liboz/Paket. The steps I did to make them are:

1. Create a new ASP.NET project.
2. Select MVC and no authentication/no cloud
3. Run `paket convert-from-nuget ` via the visual studio plugin (also did it manually and had the same issue)
4. Get the following build errors:

       Error/Warnings|  
------------- |
|The "Csc" task could not be initialized with its input parameters.|
|The "ChecksumAlgorithm" parameter is not supported by the "Csc" task. Verify the parameter exists on the task, and it is a settable public instance property.|
|Error encountered while loading the project. Some project features, such as full solution analysis for the failed project and projects that depend on it, have been disabled.|

The 3rd is actually only a warning, but the other 2 break the build.

**Comment by forki** [comment]
did you try to unpin the compiler nuget verions after that and run paket update? There was a bug in the CSC nuget package that did allow to install in path without version no. but that was fixed

**Comment by liboz** [comment]
@forki 
If I remove the compiler nuget (Microsoft.Net.Compilers) from the fresh ASP.NET application and then run the convert form nuget, everything works fine.

The fix also applies if I remove that nuget after converting (which was what gave me the broken state in the first place).

**Comment by forki** [comment]
Did you try to update the package after convert? I think it's a bug (or
maybe better incompatibility) in older C# roslyn nuget package.

**Comment by liboz** [comment]
Well I removed the package entirely and paket worked. Do you mean for me to testing trying to re-add the package through paket after the successful conversion?

**Comment by forki** [comment]
Yes please. Try to add it with latest version and see if everything still
works.

**Comment by liboz** [comment]
Yea removing the fixed version # from the paket.dependencies for Microsoft.Net.Compilers seems to have fixed it as well. That's a strange incompatibility.

**Comment by forki** [comment]
I think the issue is that nuget always have the version number in the path
and for some strange reason the csc compiler depended on that. But they
fixed that.

On Jul 31, 2016 18:55, "Libo Zeng" <notifications@github.com> wrote:

Yea removing the fixed version # from the paket.dependencies for
Microsoft.Net.Compilers seems to have fixed it as well. That's a strange
incompatibility.

—
You are receiving this because you were mentioned.

Reply to this email directly, view it on GitHub
<https://github.com/fsprojects/Paket/issues/1789#issuecomment-236441382>,
or mute the thread

**Comment by SamJongenelen** [comment]
I can confirm i have the same issue. Will try the steps to fix.

**Comment by kjellski** [comment]
I don't know wether that might help, but I ran into the same error because I tried to ´paket convert-from-nuget´ from the folder above the solution folder.

I've been **not** able to fix this with deleting the leftover in the folder above (packages folder, paket.dependencies and lock), then step into the solution folder and try ´paket convert-from-nuget --force´. Same error after a clean and build afterwards.

**Comment by forki** [comment]
?

what version of the csc nuget package do you have installed?

**Comment by hanskhe** [comment]
I am having the same issue. After updating to update 3 of VS 2015 there is a compiler error. Moving the line `<Import Project="$(MSBuildBinPath)\Microsoft.CSharp.targets" />` as described fixes the compiler error, but breaks the loading of roslyn files to the `\bin\roslyn\`-directory, causing the application to be unable to load.

As part of our .csproj we have 

```  
<PropertyGroup>
    <PostBuildEvent>if not exist "$(WebProjectOutputDir)\bin\Roslyn" md "$(WebProjectOutputDir)\bin\Roslyn"  
    start /MIN xcopy /s /y /R "$(OutDir)roslyn\*.*" "$(WebProjectOutputDir)\bin\Roslyn"</PostBuildEvent>
</PropertyGroup>
```
to ensure roslyn files are loaded. This no longer seems to work when the above line is moved.

**Comment by isaacabraham** [comment]
Is this the same as #1887 ?

**Comment by ianbattersby** [comment]
@isaacabraham Possibly, this one was also fixed by uplifting the version of imports used (I think to 1.2 and something), if the change in that uplift is that described in #1887 I'd think it is the same.
